py/src/braintrust/logger.py

```diff
@@ -1191,8 +1226,6 @@ def __next__(self):
 
 class ObjectFetcher:
     def __init__(self, object_type, pinned_version=None, mutate_record=None):
-        assert hasattr(self, "id"), "ObjectFetcher subclasses must have an 'id' attribute"
```

Comment:
Unrelated bugfix. Removing this because querying the object `id` inadvertently activates the lazy-login logic, making the lazy-login futile.